ext/sockets/sockets.c

```diff
@@ -1229,9 +1231,9 @@ PHP_FUNCTION(socket_connect)
 		case AF_INET6: {
 			struct sockaddr_in6 sin6 = {0};
 
-			if (argc != 3) {
-				php_error_docref(NULL, E_WARNING, "Socket of type AF_INET6 requires 3 arguments");
-				RETURN_FALSE;
+			if (ZEND_NUM_ARGS() != 3) {
+				zend_argument_count_error("Socket of type AF_INET6 requires port to be specified");
```

Comment:
I used an `argument_value_error` with a message similar to `must be specified for the AF_INET6 socket type` in a similar situation

Reply:
Hum, maybe? It feels like a count error is more appropriate as there is a missing argument, but could change it to a ValueError.

Reply:
My reasoning is that because this parameter is optional, and since it should be nullable in reality (and it will be when I start fixing optional params with UNKNOWN default values in stubs), we should think about this error that the default value is not appropriate in case of the aforementioned socket type. At least this is what I think 😊

Reply:
Agree with @kocsismate.